modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CJKBigramFilterFactory.java

```diff
@@ -89,4 +90,11 @@ public TokenStream create(TokenStream tokenStream) {
         return filter;
     }
 
+    @Override
+    public TokenFilterFactory getSynonymFilter(boolean lenient) {
+        if (outputUnigrams) {
+            return IDENTITY_FILTER;     // don't combine for synonyms
```

Comment:
what is the reasoning here ? If unigrams are produced we should match synonyms based on them ? Returning an idendity filter is the same as ignoring rules that produces bigram so `lenient` option is enough here ?

Reply:
Looking a bit closer at the code, unigrams are always output before bigrams, so returning `IDENTITY_FILTER` won't actually work here.  In fact, I don't think this filter can ever work with synonyms if `outputUnigrams` is true, so we should just throw an error.

Reply:
+1 to throw an error